src/instruction.rs

```diff
@@ -364,6 +448,24 @@ pub struct Arithmetic {
     pub source: ArithmeticOperand,
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub struct TernaryLogic {
+    pub operator: TernaryOperator,
+    pub operands: (MemoryReference, MemoryReference, TernaryOperand),
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct BinaryLogic {
```

Comment:
My comment above about `TernaryLogic` aside, I think `BinaryLogic` and `UnaryLogic` are more intuitive. I'm not sure if that's arbitrary or inconsistent.

That is, if I wanted an `LT`, I'd look for `Comparison`. If I wanted a `NEG`, `UnaryLogic` seems like it would make sense.

That being said, I wonder if we'd want a suite of macros where the names are the Quil instruction names themselves, so there's a part of the code that maps 1<>1 with Quil, while not requiring all users of quil-rust to match against so many variants of `Instruction` (as they would have to if `LT`, `GT`, etc got their own variants)